### PR TITLE
 Add scaffolding for less verbose error reporting

### LINKED ISSUE #54: Catch exceptions in order to give better error message
e.g. if a JSON file is invalid we get a traceback ending:

```
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
```

We could catch this and give a much better error message, without a traceback, and with useful information such as what file we were trying to open.

### PR DESCRIPTION
**Pull request by jpmckinney**
closes #54 

This builds on #42, which introduces `--verbose`. It doesn't go through the effort of improving all possible error messages. That can be addressed in specific follow-up issues.

### CODE REVIEW

**Comment by odscjames** [comment]
Just to clear (it confused me for a second), P.R. #247 should be done before this one.

File: flattentool/cli.py [hunk 1]
```diff
@@ -222,19 +230,40 @@ def main():
...
-            create_template(**kwargs_from_parsed_args(args))
-        except (OSError, IOError) as e:
-            print(text_type(e))
-            return
-    elif args.subparser_name == 'flatten':
-        flatten(**kwargs_from_parsed_args(args))
-    elif args.subparser_name == 'unflatten':
-        unflatten(**kwargs_from_parsed_args(args))
+    else:
+        def handler(type, value, traceback):
```
Comment (kindly) [inline flattentool/cli.py]: As with previous pull request would like this not nested here.

Comment (kindly) [inline flattentool/cli.py]: Would prefer this to be a top level function as well, possibly a new module for warnings customisations.

Comment (jpmckinney) [reply flattentool/cli.py]: The closure is needed here (the handler refers to `args`). If moved, we'd need a method that returns a method, which seems to increase complexity without clear benefit.

**Comment by kindly** [comment]
closed as #279 incorperated this

**Pull request closed without merging**